app/src/main/kotlin/com/softteco/template/navigation/AppNavHost.kt

```diff
@@ -201,6 +221,40 @@ fun NavGraphBuilder.homeGraph(navController: NavController, modifier: Modifier =
         composable(Screen.Notifications.route) {
             NotificationsScreen(onBackClicked = { navController.navigateUp() }, modifier)
         }
+        composable(
+            route = Screen.ThermometerDashboard.route,
+            arguments = listOf(navArgument(DEVICE_ID_KEY) { type = NavType.StringType })
+        ) { backStackEntry ->
+            val deviceId = backStackEntry.arguments?.getString(DEVICE_ID_KEY) ?: ""
```

Comment:
Wouldn't it be better to pass the argument to the ViewModel directly via SaveStateHandle?